Frameworks/Foundation/_UTTypeInternal.mm

```diff
@@ -0,0 +1,585 @@
+            //  Key: Mime Type (CFStringRef)
+            //  Value: CFMutableArray of UTI strings (CFMutableArrayRef)
+            for (const auto& mimeType : entry.mimeTypes) {
+                if (!CFDictionaryContainsKey(s_mimeTypeToSystemUTIMap, (const void*)mimeType)) {
+                    CFMutableArrayRef systemUTIArray = CFArrayCreateMutable(kCFAllocatorDefault, 0, &kCFTypeArrayCallBacks);
+                    CFArrayAppendValue(systemUTIArray, (const void*)entry.utiName);
+                    CFDictionaryAddValue(s_mimeTypeToSystemUTIMap, (const void*)mimeType, (const void*)systemUTIArray);
+                } else {
+                    CFMutableArrayRef systemUTIArray = static_cast<CFMutableArrayRef>(
+                        const_cast<void*>(CFDictionaryGetValue(s_mimeTypeToSystemUTIMap, (const void*)mimeType)));
+                    CFArrayAppendValue(systemUTIArray, (const void*)entry.utiName);
+                    CFDictionarySetValue(s_mimeTypeToSystemUTIMap, (const void*)mimeType, (const void*)systemUTIArray);
+                }
+            }
+        }
+    });
+}
+
+/**
+ * Retrieves the UTI for the given file name extension.
+ * @param fileNameExtension file name extension
+ * @return UTI
+ */
+CFArrayRef _UTGetUTIsForFileNameExtension(CFStringRef fileNameExtension) {
+    if (fileNameExtension == NULL) {
+        return NULL;
+    }
+
+    CFArrayRef UTIs =
+        static_cast<CFArrayRef>(const_cast<void*>(CFDictionaryGetValue(s_fileExtensionToSystemUTIMap, (const void*)fileNameExtension)));
+    return UTIs ? CFArrayCreateCopy(kCFAllocatorDefault, UTIs) : NULL;
+}
+
+/**
+ * Retrieves the UTI for the given MIME type.
+ * @param mimeType MIME type
+ * @return UTI
+ */
+CFArrayRef _UTGetUTIsForMIMEType(CFStringRef mimeType) {
+    CFArrayRef UTIs = static_cast<CFArrayRef>(const_cast<void*>(CFDictionaryGetValue(s_mimeTypeToSystemUTIMap, (const void*)mimeType)));
+    return CFArrayCreateCopy(kCFAllocatorDefault, UTIs);
+}
+
+/**
+ * Retrieves the first file extension associated with the given UTI.
+ * @param UTI UTI string
+ * @return file name extension associated with the UTI
+ */
+CFStringRef _UTGetFileNameExtensionForUTI(CFStringRef UTI) {
+    SystemUTIMapEntry* entry = static_cast<SystemUTIMapEntry*>(const_cast<void*>(CFDictionaryGetValue(s_systemUTIMap, (const void*)UTI)));
+    if (entry != nullptr) {
+        return entry->fileNameExtensions[0];
+    }
+
+    return NULL;
+}
+
+/**
+ * Retrieves the first MIME type associated with the given UTI.
+ * @param UTI UTI string
+ * @return MIME type associated with the UTI
+ */
+CFStringRef _UTGetMimeTypeForUTI(CFStringRef UTI) {
+    SystemUTIMapEntry* entry = static_cast<SystemUTIMapEntry*>(const_cast<void*>(CFDictionaryGetValue(s_systemUTIMap, (const void*)UTI)));
+    if (entry != nullptr) {
+        return entry->mimeTypes[0];
+    }
+
+    return NULL;
+}
+
+#pragma endregion internalMapHelpers
+
+#pragma region directFunctionalInteropMethods
+/**
+ @Status Caveat
```

Comment:
removing it and just leaving the notes

---
In reply to: [95273116](https://github.com/Microsoft/WinObjC/pull/1662#discussion_r95273116) [](ancestors = 95273116)